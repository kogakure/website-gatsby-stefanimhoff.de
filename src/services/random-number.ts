export const getRandomNumber = (max: number) =>
  String(Math.floor(Math.random() * Math.floor(max)));
