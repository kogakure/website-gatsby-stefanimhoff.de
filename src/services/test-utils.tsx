import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
