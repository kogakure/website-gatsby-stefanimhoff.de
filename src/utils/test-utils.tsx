import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';
import { ThemeType } from '../typings/Theme';
import { lightTheme } from '../themes';

interface CustomTheme {
  theme: ThemeType;
}

type RenderOptionsWithTheme = RenderOptions & CustomTheme;

const customRender = (
  node: React.ReactElement,
  options: RenderOptionsWithTheme = { theme: lightTheme }
): RenderResult => {
  const { theme } = options;
  const rendered = render(
    <ThemeProvider theme={theme}>
      <EmojifyProvider>
        <LocalizedDateProvider>{node}</LocalizedDateProvider>
      </EmojifyProvider>
    </ThemeProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (
      ui: React.ReactElement,
      rerenderOptions?: RenderOptionsWithTheme
    ) => customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
