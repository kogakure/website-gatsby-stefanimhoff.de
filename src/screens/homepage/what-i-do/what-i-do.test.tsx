import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../../services/test-utils';

import { WhatIDo } from '.';

describe('Homepage', () => {
  describe('WhatIDo', () => {
    test('renders correctly', () => {
      const { container } = render(<WhatIDo />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('is accessible', async () => {
      const { container } = render(<WhatIDo />);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });
});
