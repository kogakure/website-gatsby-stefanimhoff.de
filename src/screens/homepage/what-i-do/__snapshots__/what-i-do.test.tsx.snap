// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage WhatIDo renders correctly 1`] = `
.c1 {
  font-size: clamp(1.50rem,2.9vw,3.25rem);
  font-weight: 900;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1.0;
  margin-bottom: clamp(1.44rem,3.70vw,1.728rem);
  margin-top: 0;
}

.c1 .autolink {
  background: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: var(--colorForeground);
  margin-left: clamp( 0.335rem,0.72vw,0.402rem);
  opacity: 0;
  vertical-align: middle;
}

.c1:hover .autolink,
.c1:focus .autolink {
  opacity: 0.5;
}

.c3 {
  font-size: clamp(1.00rem,1.1vw,1.25rem);
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1.6;
  margin-bottom: clamp(1.44rem,3.70vw,1.728rem);
  margin-top: 0;
}

.c5 {
  color: var(--colorForeground);
  font-weight: 600;
  -webkit-text-decoration-color: var(--colorForegroundSoft);
  text-decoration-color: var(--colorForegroundSoft);
  -webkit-text-decoration-thickness: 0.1em;
  text-decoration-thickness: 0.1em;
  text-underline-offset: 0.1em;
}

.c5:hover,
.c5:focus {
  -webkit-text-decoration-color: var(--colorAccent);
  text-decoration-color: var(--colorAccent);
}

.c4 {
  color: var(--colorForeground);
  font-weight: 600;
  -webkit-text-decoration-color: var(--colorForegroundSoft);
  text-decoration-color: var(--colorForegroundSoft);
  -webkit-text-decoration-thickness: 0.1em;
  text-decoration-thickness: 0.1em;
  text-underline-offset: 0.1em;
}

.c4:hover,
.c4:focus {
  -webkit-text-decoration-color: var(--colorAccent);
  text-decoration-color: var(--colorAccent);
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  fill: var(--colorForeground);
  margin-left: 0.2em;
  position: relative;
  top: 0.2em;
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

a:hover .c6,
a:active .c6,
a:focus .c6 {
  -webkit-transform: translate3D(0.25rem,0,0);
  -ms-transform: translate3D(0.25rem,0,0);
  transform: translate3D(0.25rem,0,0);
}

.c0 {
  grid-column: 2 / -2;
  grid-row: 4;
}

.c2 {
  -webkit-column-count: 1;
  column-count: 1;
  -webkit-column-gap: 5.55vw;
  column-gap: 5.55vw;
}

@media screen and (prefers-reduced-motion:no-preference) {
  .c1 .autolink {
    -webkit-transition-duration: 500ms;
    transition-duration: 500ms;
    -webkit-transition-property: opacity;
    transition-property: opacity;
    -webkit-transition-timing-function: ease-in-out;
    transition-timing-function: ease-in-out;
  }
}

@media (min-width:768px) {
  .c0 {
    grid-column: 3 / -3;
    grid-row: 3;
  }
}

@media (min-width:1200px) {
  .c0 {
    grid-column: 5 / -5;
  }
}

@media (min-width:768px) {
  .c2 {
    -webkit-column-count: 2;
    column-count: 2;
  }
}

<div
  class="c0"
>
  <h2
    class="c1"
  >
    What I Do
  </h2>
  <div
    class="c2"
  >
    <p
      class="c3"
    >
      I create and build beautiful, user-centric layouts for devices of all sizes on the web for nearly two decades. I love bringing simplicity and minimalism to everything I create.
    </p>
    <p
      class="c3"
    >
      I deeply care about good user experience: Writing good text, choosing the right photography, typography, and colours. Programming fast, reusable, well-tested and easy-to-maintain code or creating engaging animations.
    </p>
    <p
      class="c3"
    >
      I am currently working at
       
      <a
        class="c4"
        href="https://www.xing.com/"
        rel="nofollow noopener noreferrer external"
      >
        XING
      </a>
      , the biggest business network in Europe, working on the Design System.
    </p>
    <p
      class="c3"
    >
      I love innovating: During multiple
       
      <a
        class="c4"
        href="https://medium.com/xing-hackweek"
        rel="nofollow noopener noreferrer external"
      >
        HackWeeks
      </a>
       I was part of a small team creating a complete new job platform for freelancers. I am also a three-time winner of the
       
      <em>
        Prototyping Days
      </em>
      , a yearly two-day company-wide prototyping event.
    </p>
    <a
      block="0"
      class="c5"
      href="/projects/"
    >
      More
      <span
        class="c6"
      >
        <svg
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M16.172 11l-5.364-5.364 1.414-1.414L20 12l-7.778 7.778-1.414-1.414L16.172 13H4v-2z"
          />
        </svg>
      </span>
    </a>
  </div>
</div>
`;
